src/ctia/flows/crud.clj

```diff
@@ -7,20 +7,35 @@
             [ctia.flows.hooks :as h]
             [ctia.events.obj-to-event :refer [to-create-event
                                               to-update-event
-                                              to-delete-event]]))
+                                              to-delete-event]]
+            [ring.util.http-response :as http-response]))
 
-(defn find-id
+(defn- find-id
   "Lookup an ID in a given entity.  Parse it, because it might be a
    URL, and return the short form ID.  Returns nil if the ID could not
    be found."
   [{id :id}]
   (when (seq id)
     (id/str->short-id id)))
 
-(defn make-id
+(defn- make-id
   [entity-type]
   (str (name entity-type) "-" (UUID/randomUUID)))
 
+(defn- find-entity-id [identity entity-type entity prev-entity]
+  (or (find-id prev-entity)
+      (when-let [entity-id (find-id entity)]
+        (when-not (auth/capable? identity :specify-id)
+          (throw (http-response/bad-request!
+                  {:error "Missing capability to specify entity ID"
+                   :entity entity})))
+        (if (id/valid-short-id? entity-id)
```

Comment:
I thought about it, and I agree, it looks nicer.  I ended up not doing it because I was afraid it makes the entity-id value harder to spot when scanning the code.  Your suggestion is probably better though.